src/game/BattleGround/BattleGround.cpp

```diff
@@ -583,6 +583,9 @@ void BattleGround::CastSpellOnTeam(uint32 SpellID, Team teamId)
 
 void BattleGround::RewardHonorToTeam(uint32 Honor, Team teamId)
 {
+    if (teamId == TEAM_NONE)
+        return;
```

Comment:
Instead of adding this check in the one required place this now adds unnecessary checks for a team in cases where we already knew that the team is not none and makes it harder harder to reason about the code

Reply:
no team gets no RewardHonorToTeam, given this call is made from EndBattleGround and isn't very intensive it doesn't seem like a reasonable optimization.

Reply:
Obviously the performance is negligible in all touched code but why intentionally write more inefficient and harder to understand code than my suggestion